src/model/test/HeatPumpAirToWaterFuelFiredCooling_GTest.cpp

```diff
@@ -0,0 +1,271 @@
+  EXPECT_EQ(2.0, hp.nominalCOP());
+  ASSERT_TRUE(hp.designFlowRate());
+  EXPECT_EQ(3.0, hp.designFlowRate().get());
+  EXPECT_FALSE(hp.isDesignFlowRateAutosized());
+  EXPECT_EQ(4.0, hp.designSupplyTemperature());
+  ASSERT_TRUE(hp.designTemperatureLift());
+  EXPECT_EQ(5.0, hp.designTemperatureLift().get());
+  EXPECT_FALSE(hp.isDesignTemperatureLiftAutosized());
+  EXPECT_EQ(6.0, hp.sizingFactor());
+  EXPECT_EQ("ConstantFlow", hp.flowMode());
+  EXPECT_EQ("WetBulb", hp.outdoorAirTemperatureCurveInputVariable());
+  EXPECT_EQ("LeavingEvaporator", hp.waterTemperatureCurveInputVariable());
+  EXPECT_EQ(curve1.handle(), hp.normalizedCapacityFunctionofTemperatureCurve().handle());
+  EXPECT_EQ(curve2.handle(), hp.fuelEnergyInputRatioFunctionofTemperatureCurve().handle());
+  EXPECT_EQ(curve3.handle(), hp.fuelEnergyInputRatioFunctionofPLRCurve().handle());
+  EXPECT_EQ(0.5, hp.minimumPartLoadRatio());
+  EXPECT_EQ(0.75, hp.maximumPartLoadRatio());
+  ASSERT_TRUE(hp.cyclingRatioFactorCurve());
+  EXPECT_EQ(curve4.handle(), hp.cyclingRatioFactorCurve().get().handle());
+  EXPECT_EQ(12.0, hp.nominalAuxiliaryElectricPower());
+  ASSERT_TRUE(hp.auxiliaryElectricEnergyInputRatioFunctionofTemperatureCurve());
+  EXPECT_EQ(curve5.handle(), hp.auxiliaryElectricEnergyInputRatioFunctionofTemperatureCurve().get().handle());
+  ASSERT_TRUE(hp.auxiliaryElectricEnergyInputRatioFunctionofPLRCurve());
+  EXPECT_EQ(curve6.handle(), hp.auxiliaryElectricEnergyInputRatioFunctionofPLRCurve().get().handle());
+  EXPECT_EQ(13.0, hp.standbyElectricPower());
+
+  hp.autosizeNominalCoolingCapacity();
+  hp.autosizeDesignFlowRate();
+  hp.autosizeDesignTemperatureLift();
+
+  EXPECT_FALSE(hp.nominalCoolingCapacity());
+  EXPECT_TRUE(hp.isNominalCoolingCapacityAutosized());
+  EXPECT_FALSE(hp.designFlowRate());
+  EXPECT_TRUE(hp.isDesignFlowRateAutosized());
+  EXPECT_FALSE(hp.designTemperatureLift());
+  EXPECT_TRUE(hp.isDesignTemperatureLiftAutosized());
+
+  hp.resetCompanionHeatingHeatPump();
+  hp.resetEndUseSubcategory();
+  hp.resetNominalCoolingCapacity();
+  hp.resetDesignFlowRate();
+  hp.resetCyclingRatioFactorCurve();
+  hp.resetAuxiliaryElectricEnergyInputRatioFunctionofTemperatureCurve();
+  hp.resetAuxiliaryElectricEnergyInputRatioFunctionofPLRCurve();
+
+  EXPECT_FALSE(hp.companionHeatingHeatPump());
+  EXPECT_EQ("General", hp.endUseSubcategory());
+  EXPECT_TRUE(hp.isEndUseSubcategoryDefaulted());
+  EXPECT_FALSE(hp.nominalCoolingCapacity());
+  EXPECT_FALSE(hp.designFlowRate());
+  EXPECT_FALSE(hp.cyclingRatioFactorCurve());
+  EXPECT_FALSE(hp.auxiliaryElectricEnergyInputRatioFunctionofTemperatureCurve());
+  EXPECT_FALSE(hp.auxiliaryElectricEnergyInputRatioFunctionofPLRCurve());
+}
+
+TEST_F(ModelFixture, HeatPumpAirToWaterFuelFiredCooling_remove) {
+  Model m;
+  HeatPumpAirToWaterFuelFiredCooling hp(m);
+  auto size = m.modelObjects().size();
+  EXPECT_FALSE(hp.remove().empty());
+  EXPECT_EQ(size - 1, m.modelObjects().size());
```

Comment:
I consistently get confused about how resources objects are treated... That means we create like 10 curves in the Ctor, remove the HP, and we get 10 orphaned curves there?